### PR TITLE
Data list view: Make filter row, table header, and pagination sticky

### PR DESCRIPTION
**Pull request by jameskoster**
## What?
In data list views, try making the filter row, table header, and pagination row sticky.

## Why?
These tools can be useful regardless of scroll position. If you're displaying 100 records per row a lot of scrolling is required to access them.

## Testing Instructions
1. In the Site editor navigate to Pages > Manage all pages
2. Check that the ~filter row~, column headers, and pagination stays in view as you scroll

https://github.com/WordPress/gutenberg/assets/846565/d077a762-0cf6-4bea-b1fa-4ecc3a8db40e

### CODE REVIEW

**Comment by jasmussen** [comment]
Worth trying.

How much do we need to make sticky, though? What would this look like on a tiny 12 inch laptop? I.e. do we need the Pages heading to be sticky as well? (Entirely separate but related question, do we need the Pages heading to be visible at all if there's a Pages heading in the sidebar?)

**Comment by jameskoster** [comment]
Good questions, we should definitely be diligent here. I think pagination speaks for itself. Column headings seem useful so that you can always interpret the data in the column. That's especially important if you had similar data points (e.g. Publish date and Last modified). Perhaps the filter row / view options don't need to stick since you can filter from the column heading dropdowns. What do you think?

The view title is already sticky, but with regards to your separate question, I opened an issue [here](https://github.com/WordPress/gutenberg/issues/56199) about that earlier today.

**Comment by jameskoster** [comment]
I pushed an update to unstick the filter row, and updated the video in the OP.

File: packages/edit-site/src/components/dataviews/style.scss [hunk 1]
```diff
@@ -11,6 +11,15 @@
 
 .dataviews-pagination {
 	margin-top: auto;
+	position: sticky;
+	bottom: 0;
+	background-color: $white;
+	padding: $grid-unit-20 0;
+	border-top: 1px solid $gray-200;
```
Comment (jasmussen) [inline packages/edit-site/src/components/dataviews/style.scss]: Nit, can be $border-width

File: packages/edit-site/src/components/dataviews/style.scss [hunk 2]
```diff
@@ -37,6 +47,21 @@
...
+	}
+	thead {		
+		tr {
+			border: 0;
+		}
+		th {
+			position: sticky;
+			top: 0;
+			background-color: $white;
+			box-shadow: inset 0 -1px 0 $gray-200;
```
Comment (jasmussen) [inline packages/edit-site/src/components/dataviews/style.scss]: Maybe -$border-width?

File: packages/edit-site/src/components/dataviews/style.scss [hunk 3]
```diff
@@ -37,6 +47,21 @@
 	}
 	tr {
 		border-bottom: 1px solid $gray-100;
```
Comment (jasmussen) [inline packages/edit-site/src/components/dataviews/style.scss:49]: Perhaps $border-width while you're here :D

**Review by jasmussen — APPROVED** [review]
Worth trying, especially useful for shorter screens.

**Comment by oandregal** [comment]
Nice one!

>  I.e. do we need the Pages heading to be sticky as well? (Entirely separate but related question, do we need the Pages heading to be visible at all if there's a Pages heading in the sidebar?)

I think we could do with actually removing the page title, I've prepared a PR before seeing this one https://github.com/WordPress/gutenberg/pull/56241 The title seems to me a bit redundant with the sidebar info (hierarchy/status: "Pages in draft").

**Comment by afercia** [comment]
I just noticed this PR introduced two important usability and accessibility problems that need to be fixed:
1. When navigating the table content with the keyboard, the currently focused element may be entirely hidden behind the sticky header / footer. This is a violation of the WCAG 2.2 [Focus not obscured](https://www.w3.org/TR/WCAG22/#focus-not-obscured-minimum) success criterium and needs to be fixed.
2. The button tooltips show on toop of the sticky header / footer when scrolling the table.

I will create a new issue.

**Pull request merged**